django/db/models/query.py

```diff
@@ -1152,18 +1152,21 @@ def _chain(self, **kwargs):
         obj.__dict__.update(kwargs)
         return obj
 
+    _clone_attrs = ('_sticky_filter', '_for_write', '_prefetch_related_lookups', '_iterable_class', '_fields')
+
     def _clone(self):
         """
         Return a copy of the current QuerySet. A lightweight alternative
         to deepcopy().
         """
         c = self.__class__(model=self.model, query=self.query.chain(), using=self._db, hints=self._hints)
-        c._sticky_filter = self._sticky_filter
-        c._for_write = self._for_write
-        c._prefetch_related_lookups = self._prefetch_related_lookups[:]
+        for attr in self._clone_attrs:
+            try:
+                value = self.__dict__[attr]
+            except KeyError:
+                continue
+            setattr(c, attr, value)
```

Comment:
This could probably use a series of `if` like `Query.clone` does.

Reply:
`c.__dict__ = self.__dict__.copy()` like `Query._clone` would be ideal but I guess we can't have it here due to subclasses?

a series of `if`s may or may not be faster...

Reply:
Right, we can't use the `c = Empty(); c.__class__ = self.__class__;` trick that `Query` does here because it might break `QuerySet` sublasses expecting `__init__` to be called on `_clone()`. I guess we could break this assumption but what I like about this patch is that it essentially maintains backward compatibility.

Reply:
Maintaining backwards compat ftw. Let's keep as-is, should be a good boost anyway.